docs/index.md

```diff
@@ -167,9 +167,141 @@ Returning:
+
+There's three logical varieties of slots:
+
+- [Delegate Slots](#delegate-slots) - Used to render other components.
+- [Lambda Slots](#lambda-slots) - Used to render strings or initialized
+- components.
+- [Pass through Slots](#pass-through-slots) - Used to pass content directly to a
+    component.
+
+##### Delegate Slots
+
+Delegate slots are simple slot definitions that delegate to another component.
+
+`# blog_component.rb`
+
+```ruby
+class BlogComponent < ViewComponent::Base
+  # `HeaderComponent` is nested inside of this component, we have to
+  # define it as a string instead of a class name.
+  renders_one :header, "HeaderComponent"
+
+  # `PostComponent` is defined in another file, so we can refer to it by class
+  # name without using a string.
+  renders_many :post, PostComponent
+
+  class HeaderComponent < ViewComponent::Base
+    attr_reader :title
+
+    def initialize(title:)
+    end
+  end
+end
+```
+
+`# blog_component.html.erb`
+
+```erb
+<div>
+  <h1><%= header %></h1> <!-- render the header component -->
+
+  <% posts.each do |post| %>
+    <div class="blog-post-wrapper">
+      <%= post %> <!-- render an individual post -->
+    </div>
+  <% end %>
+</div>
+```
+
+`# index.html.erb`
+
+```erb
+<%= render BlogComponent.new do |c| %>
+  <% c.header do %>
+    <%= link_to "My Site", root_path %>
+  <% end %>
+
+  <%= c.post(title: "My blog post") do %>
+    Really interesting stuff.
+  <% end %>
+
+  <%= c.post(title: "Another post!") do %>
+    Blog every day.
+  <% end %>
+<% end %>
+```
+
+##### Labmda Slots
+
+Lambda slots render their return value on the page. This is useful for creating
+small slots via helpers like `content_tag` or using returning components with
+parent-component specific default values.
+
+```ruby
+class Blogcomponent < ViewComponent::Base
+  # Renders the returned string
+  renders_one :header, -> (title:) do
+    content_tag :h1 do
+      link_to "My Blog", root_path
+    end
+  end
+
+  # Returns a component that will be rendered in that slot with a default
+  # argument.
+  renders_many :posts, -> (title:, classes:) do
+    PostComponent.new(title: title, classes: "my-default-class " + classes)
+  end
+end
+```
+
+##### Pass Through Slots
+
+Pass through slots simply render the content given the block.
+
+Pass through slots can be defined by omitting the second argument to
+`renders_one` and `renders_many`.
+
+`# blog_component.rb`
+
+```ruby
+class BlogComponnt < ViewComponent::Base
+  renders_one :header
+end
+```
+
+`# blog_component.html.erb`
+
+```ruby
+<div>
+  <%= render BlogComponent.new do |c| %>
+    <%= c.title do %>
+      <%= link_to "My blog", root_path %>
+    <% end %>
+  <% end %>
+</div>
+```
+
+#### Slots V1 (deprecated)
+
+_Slots V1 are now deprecated and will be removed in 3.0. Please migrate to [Slots V2](#slots)_
 
 Slots enable multiple blocks of content to be passed to a single ViewComponent, reducing the need for sub-components (e.g. ModalHeader, ModalBody).
```

Comment:
Question: should this section really be in the main README in full length? Or could we move it to something like `UPGRADING_FROM_SLOTS_V1.md` to reduce confusion.

Reply:
Good question. I was thinking that it would make more sense as a separate docs file, but not sure how that would work with viewcomponent.org